### PR TITLE
Fix broken stories, misc. project improvements

### PR DESCRIPTION
**Pull request by egargan**
This PR fixes broken stories for the canvas map and SVG map components, as well as addressing a couple of build and workflow issues.

## SVG Map

Each of the SVG map component's "layers" stories have been fixed by tweaking the story config's `decorate` and `render` properties.

After discussing with Niels, it was decided that the SVG map component does not need its _zoom_ functionality, and so this PR removes all related code from this component.

This also lead to the creation of the `useThrowIfNonLayerChildren` hook, which effectively replaces the `useOrganisedChildren` hook's "throw an error if we get an unrecognised child component" logic.

## `checkJs: true`

We'll want this enabled so that our build process does type checking in our JS files, and enables warnings, errors, completions etc. in our editors.

## `jsconfig.json` paths

We configure our import aliases in our Vite config currently, but not in our `jsconfig.json`. This means we can't do things like "go to definition" in our editors, and whatever we import won't be type checked.

We could just copy over our Vite config aliases into our jsconfig file, but to avoid duplication, we can use a Vite plugin to parse and copy the aliases from our jsconfig into our Vite config.

### CODE REVIEW

File: jsconfig.json [hunk 1]
```diff
@@ -1,14 +1,25 @@
...
     "module": "es2022",
     "esModuleInterop": true,
     "target": "ESNext",
     "jsx": "react-jsx",
     "jsxImportSource": "preact",
     "jsxFactory": "createElement",
     "skipLibCheck": true,
-    "checkJs": false
+    "checkJs": true,
+    "allowJs": true
```
Comment (njdehoog) [inline jsconfig.json:23]: Nice work on this 👍 

Minor thing: `allowJS` is true by default in jsconfig

Screenshot 2024-08-07 at 08 11 26

https://code.visualstudio.com/docs/languages/jsconfig

File: src/lib/components/molecules/canvas-map/Map.jsx [hunk 2]
```diff
@@ -75,7 +75,9 @@ export const Map = forwardRef(
       }
 
       return () => {
-        ref.current = null
+        if (ref) {
+          ref.current = null
+        }
```
Comment (njdehoog) [inline src/lib/components/molecules/canvas-map/Map.jsx:80]: Ha, you ran into the same bug that I did. Good catch.

File: src/lib/components/molecules/svg-map/hooks/useThrowIfNonLayerChildren.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+      (child) => !layerTypes.includes(child.type),
+    )
+
+    if (firstNonLayerChild) {
+      throw Error(
+        `unknown component type passed as child to SVGMap component: ${firstNonLayerChild.type}`,
+      )
+    }
+  }, [children])
+}
```
Comment (njdehoog) [inline src/lib/components/molecules/svg-map/hooks/useThrowIfNonLayerChildren.js:22]: 🎉 👍

**Review by njdehoog — APPROVED** [review]
Great work! 💯

File: jsconfig.json (hunk 1 above)
Comment (egargan) [reply jsconfig.json:23]: Ah good to know, thanks!

**Pull request merged**